tests/core.py

```diff
@@ -1053,6 +1054,61 @@ def setUp(self):
             dag_folder=DEV_NULL, include_examples=True)
         # Persist DAGs
 
+    def test_pools_add_pool_if_not_exist(self):
+        session = mock.MagicMock()
+        session.add = mock.MagicMock()
+
+        pools_config = {
+            'pool1': {
+                'slot_counts': 1,
+                'description': 'haha',
+            }
+        }
+
+        cli._pools(session, pools_config)
+
+        self.assertEqual('pool1', session.add.call_args_list[0][0][0].pool)
+
+    def test_pools_delete_pool_if_not_defined(self):
+        session = mock.MagicMock()
+        session.delete = mock.MagicMock()
+        query = session.query.return_value
+        query.all.return_value = [
```

Comment:
I like these tests, but there's a lot of boilerplate related to setting up the mocks that we could abstract out. Maybe something like this?

```
def _expect_session_returns(self, existing_pools):
  session = mock.MagicMock()
  session.delete = mock.MagicMock()
  session.add = mock.MagicMock()
  query = session.query.return_value
  query.all.return_value = existing_pools
  return session

def _assert_pool_was_deleted(self, pool_name)
  self.assertEqual(pool_name, session.delete.call_args_list[0][0][0].pool)
```

And now the test becomes highly readable:
```
def test_pools_delete_pool_if_not_defined(self):
  session = self._expect_session_returns(Pool(pool='pool1'...))
  cli._pools(session, {})
  self._assert_pool_was_deleted('pool1')
```

Reply:
Agree that tests should also be readable